collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_pg.py

```diff
@@ -448,8 +448,8 @@ def rename_exists(module, array):
 def update_pgroup(module, array):
     """Update Protection Group"""
     changed = renamed = False
+    api_version = array._list_available_rest_versions()
     if module.params["target"]:
-        api_version = array._list_available_rest_versions()
```

Comment:
You cannot move this here as it is required in line 661 which is outside this `if` section

Reply:
Maybe I am mistaken but please have a closer look to indentation inside this function. Line 661 ( if RETENTION_LOCK_VERSION in api_version: ) is on the same level as line 451 ( if module.params["target"]: ).
So the variable declaration has to be before 451.
Otherwise it will not be set if there is no value in module.params["target"].